crates/rollup-boost/src/flashblocks/inbound.rs

```diff
@@ -0,0 +1,58 @@
+use super::primitives::FlashblocksPayloadV1;
+use futures::StreamExt;
+use tokio::sync::mpsc;
+use tokio_tungstenite::{connect_async, tungstenite::Message};
+use tracing::{error, info};
+use url::Url;
+
+pub struct FlashblocksReceiverService {
+    url: Url,
+    sender: mpsc::Sender<FlashblocksPayloadV1>,
+}
+
+impl FlashblocksReceiverService {
+    pub fn new(
+        url: Url,
+        sender: mpsc::Sender<FlashblocksPayloadV1>,
+    ) -> Result<Self, url::ParseError> {
+        Ok(Self { url: url, sender })
+    }
+
+    pub async fn run(self) {
+        loop {
+            match self.connect_and_handle().await {
+                Ok(()) => break,
+                Err(e) => {
+                    error!(
+                        message = "Flashblocks receiver connection error, retrying in 5 seconds",
+                        error = %e
+                    );
+                    tokio::time::sleep(std::time::Duration::from_secs(5)).await;
+                }
```

Comment:
Nit: consider simplifying to
```suggestion
        if let Err(e) = self.connect_and_handle().await {
            error!("Flashblocks receiver connection error, retrying in 5 seconds: {e}");
            tokio::time::sleep(Duration::from_secs(5)).await;
        } else {
            break;
        }
```